File: Club12-Backend/Domain/Enums/MatchStatus.cs
namespace Domain.Enums;

/// <summary>
/// The lifecycle state of a match's result, distinct from the boolean Match.IsFinished flag.
/// </summary>
public enum MatchStatus
{
    /// <summary>
    /// The default state: the fixture exists but no result has been loaded yet.
    /// </summary>
    Scheduled,

    /// <summary>
    /// The match was played and a decisive result was loaded.
    /// </summary>
    Played,

    /// <summary>
    /// The match was postponed or suspended and has no result.
    /// </summary>
    Suspended,

    /// <summary>
    /// One team did not show up, so the regulation default result was applied to the present team.
    /// </summary>
    WalkOver,

    /// <summary>
    /// The match will never be played because its tournament was canceled or force-closed while it was still pending.
    /// </summary>
    Canceled
}

File: Club12-WebClient/src/modules/core/enum/match/matchStatus.ts
/**
 * The lifecycle status of a match (HU-69/HU-73). Mirrors the backend
 * `Domain.Enums.MatchStatus` and is serialized as a string on the match
 * response DTOs.
 * @enum MatchStatus
 */
export enum MatchStatus {
  /**
   * The match has a fixture but no result loaded yet.
   */
  Scheduled = 'Scheduled',

  /**
   * A normal result was loaded (HU-69); the match has a winner.
   */
  Played = 'Played',

  /**
   * The match was suspended (HU-68/HU-73) and awaits rescheduling.
   */
  Suspended = 'Suspended',

  /**
   * A walkover was applied (HU-73): the present team was awarded the
   * regulation default result. Distinguishable from a normal `Played`.
   */
  WalkOver = 'WalkOver',

  /**
   * The match will never be played: its tournament was canceled, or
   * force-closed as finished while the match was still pending.
   */
  Canceled = 'Canceled',
}
